sdk/ai/azure-ai-generative/azure/ai/generative/synthetic/simulator/_model_tools/output_parsing.py

```diff
@@ -75,66 +87,72 @@ def flatten_outputs(
+        output_list = sorted(output_list, key=lambda x: x["input_idx"])
         for example_obj in output_list:
-            f_out.write(json.dumps(example_obj, quote_keys=True) + '\n')
+            f_out.write(json.dumps(example_obj, quote_keys=True) + "\n")
 
 
-def decode_example(
-    example: str,
-    label_keys: List[str],
-    encoding: Encoding = Encoding.JSON
-) -> Dict[str, Any]:
-    '''
+def decode_example(example: str, label_keys: List[str], encoding: Encoding = Encoding.JSON) -> Dict[str, Any]:
+    """
     Decode example from an encoding format.
 
-    Args:
-        example (str): example to decode
-        label_keys (List[str]): list of label keys to check for
-        encoding (Encoding): encoding format to use
-    Returns:
-        Dict[str, Any]: decoded example
-    '''
+    :param example: The example to decode.
+    :type example: str
+    :param label_keys: List of label keys to check for.
+    :type label_keys: List[str]
+    :param encoding: The encoding format to use.
+    :type encoding: Encoding
+    :return: The decoded example.
+    :rtype: Dict[str, Any]
+    """
     example = example.strip()
     if encoding == Encoding.JSON:
         return try_decode_json(example, label_keys)
-    elif encoding == Encoding.XML:
-        raise NotImplementedError('XML encoding not implemented.')
-    else:
-        raise ValueError(f'Unknown encoding {encoding}.')
+    if encoding == Encoding.XML:
+        raise NotImplementedError("XML encoding not implemented.")
+    raise ValueError(f"Unknown encoding {encoding}.")
 
 
 def try_decode_json(example: str, label_keys: List[str]) -> Dict[str, Any]:
-    '''Try to decode an example in a JSON encoding.'''
-    start = example.find('{')
+    """
+    Try to decode an example in a JSON encoding.
+
+    :param example: The example to decode.
+    :type example: str
+    :param label_keys: List of label keys to check for.
+    :type label_keys: List[str]
+    :return: The decoded example.
+    :rtype: Dict[str, Any]
+    """
+    start = example.find("{")
     end_index = start + 1
     last_error = None
-    
-    while -1 < (end_index := example.find('}', end_index + 1)) < len(example):
+
+    while -1 < (end_index := example.find("}", end_index + 1)) < len(example):
         try:
-            example_dict = dict(json.loads(example[start:end_index + 1]))
+            example_dict = dict(json.loads(example[start : end_index + 1]))
 
             # check if any label keys are in example
-            assert any(label_key in example_dict for label_key in label_keys), \
-                f"Failed to decode example.  No label keys found in example: {example_dict}"
+            assert any(
+                label_key in example_dict for label_key in label_keys
+            ), f"Failed to decode example.  No label keys found in example: {example_dict}"
 
             return example_dict
-        except Exception as e:
```

Comment:
Let's leave this one as a bare Exception

Reply:
fixed